### ISSUE TITLE
static worker reregistration broken

### ISSUE THREAD

**Issue by petemoore** [taskcluster/taskcluster#3476]
**Describe the bug**

taskcluster CI mac worker not running:

```
2020/09/07 12:36:14 Could not register worker: 
CALL SUMMARY
============
Method: POST
Service: community-tc.services.mozilla.com:
Response Body:
{
  "code": "ResourceNotFound",
  "message": "Worker proj-taskcluster/63-135-170-250 in worker pool proj-taskcluster/gw-ci-macos does not exist\n\n---\n\n* method:     registerWorker\n* errorCode:  ResourceNotFound\n* statusCode: 404\n* time:       2020-09-07T16:36:14.796Z",
  "requestInfo": {
    "method": "registerWorker",
    "params": {},
    "payload": "(OMITTED)",
    "time": "2020-09-07T16:36:14.796Z"
  }
}
Attempts: 1
(Permanent) HTTP response code 404
HTTP/2.0 404 Not Found
Content-Length: 417
Access-Control-Allow-Headers: X-Requested-With,Content-Type,Authorization,Accept,Origin,Cache-Control
Access-Control-Allow-Methods: OPTIONS,GET,HEAD,POST,PUT,DELETE,TRACE,CONNECT
Access-Control-Allow-Origin: *
Access-Control-Max-Age: 900
Access-Control-Request-Method: *
Alt-Svc: clear
Cache-Control: no-store no-cache must-revalidate
Content-Security-Policy: report-uri /__cspreport__;default-src 'none';frame-ancestors 'none';
Content-Type: application/json; charset=utf-8
Date: Mon, 07 Sep 2020 16:36:14 GMT
Etag: W/"1a1-bOXzXQYn0eQcuJXzWBXproKhIVU"
Server: openresty
Strict-Transport-Security: max-age=31536000
Via: 1.1 google
X-Content-Type-Options: nosniff
X-For-Request-Id: 1e020687-bf25-4d72-94b8-df675ba78a1d
X-For-Trace-Id: 33976c6a885b8dc7355353477203ca69

{
  "code": "ResourceNotFound",
  "message": "Worker proj-taskcluster/63-135-170-250 in worker pool proj-taskcluster/gw-ci-macos does not exist\n\n---\n\n* method:     registerWorker\n* errorCode:  ResourceNotFound\n* statusCode: 404\n* time:       2020-09-07T16:36:14.796Z",
  "requestInfo": {
    "method": "registerWorker",
    "params": {},
    "payload": "(OMITTED)",
    "time": "2020-09-07T16:36:14.796Z"
  }
}
administrators-Mac-mini:~ root# 
```

**Comment by petemoore** [comment]
Worker pool [proj-taskcluster/gw-ci-macos](https://community-tc.services.mozilla.com/worker-manager/proj-taskcluster%2Fgw-ci-macos) exists.

**Comment by petemoore** [comment]
Worker is [known](https://community-tc.services.mozilla.com/provisioners/proj-taskcluster/worker-types/gw-ci-macos/workers/proj-taskcluster/63-135-170-250) to the queue.

**Comment by petemoore** [comment]
I suspect worker reregistration is broken somehow. We had this before, and I still had my old go program to fix it. I thought it had been fixed, but it looks to still be an open issue.

This is the go program I've just run, to recreate the static secret, and I'm rebooting the worker. Hopefully it will start claiming tasks again soon.

```go
package main

import (
	"encoding/json"
	"fmt"
	"log"
	"time"

	tcclient "github.com/taskcluster/taskcluster/v29/clients/client-go"
	"github.com/taskcluster/taskcluster/v29/clients/client-go/tcworkermanager"
)

func main() {
	wm := tcworkermanager.NewFromEnv()
	err := wm.RemoveWorker("proj-taskcluster/gw-ci-macos", "proj-taskcluster", "63-135-170-250")
	if err != nil {
		log.Printf("Warning! %v", err)
	}
	spt := tcworkermanager.StaticProviderType{
		StaticSecret: "*************************************", // replace this with real static secret
	}
	s, err := json.Marshal(spt)
	if err != nil {
		log.Fatalf("Error! %v", err)
	}
	wcr := &tcworkermanager.WorkerCreationRequest{
		Capacity:     1,
		Expires:      tcclient.Time(time.Now().AddDate(0, 0, 30)),
		ProviderInfo: json.RawMessage(s),
	}
	wpfd, err := wm.CreateWorker("proj-taskcluster/gw-ci-macos", "proj-taskcluster", "63-135-170-250", wcr)
	if err != nil {
		log.Fatalf("Error! %v", err)
	}
	resp, err := json.MarshalIndent(wpfd, "", "  ")
	if err != nil {
		log.Fatalf("Error! %v", err)
	}
	fmt.Println(string(resp))
}
```

**Comment by petemoore** [comment]
It seems to be [claiming tasks](https://community-tc.services.mozilla.com/provisioners/proj-taskcluster/worker-types/gw-ci-macos/workers/proj-taskcluster/63-135-170-250) again.

However, I'll leave this issue open, as this is just a workaround. The static workers shouldn't need kick-starting once per month, so something appears to still be broken with static worker reregistration...

**Comment by petemoore** [comment]
CC @djmitche

**Comment by djmitche** [comment]
The worker is expiring after 30 days:

```
		Expires:      tcclient.Time(time.Now().AddDate(0, 0, 30)),
```

I believe that can be set to "forever" now, as it is no longer tied to the expiration of the worker's temporary credentials.  I think that change has been deployed (but don't quote me on it).

This was fixed in #3352 / https://github.com/taskcluster/taskcluster/commit/b0bb298d9579addf276257af4412dd292d863b2f which isn't deployed anywhere yet.  Hopefully that will be deployed in the next 30 days and we won't have to worry about this anymore.
